# 

Abstract  
We introduce a novel statistical method, PyClone, for inference of clonal population structures in cancers. PyClone is a Bayesian clustering method for grouping sets of deeply sequenced somatic mutations into putative clonal clusters while estimating their cellular prevalences and accounting for allelic imbalances introduced by segmental copy number changes and normal cell contamination. Single cell sequencing validation demonstrates that PyClone infers accurate clustering of mutations that co-occur in individual cells.  
摘要：我们引入了一种新的统计方法，PyClone，用于推断癌症中的克隆群体结构。PyClone是一种贝叶斯聚类方法，用于将一系列深度测序的体细胞突变分组为假定的克隆聚类，同时估计其细胞发生率并考虑由节段拷贝数变化和正常细胞污染引入的等位基因不平衡。单细胞测序验证表明，PyClone推断出在单细胞中共同发生的突变的准确聚类。

coverage >1000x；

Our approach uses the measurement of allelic prevalence to estimate the proportion of tumour cells harbouring a mutation (referred to herein as the ’cellular prevalence’). 
使用等位基因发病率的测量来估计携带突变的肿瘤细胞的比例；


#

https://github.com/aroth85/pyclone
https://bitbucket.org/aroth85/pyclone/wiki/Usage

## Usage
创建三个输入文件：sample1.tsv, sample2.tsv, sample3.tsv；





