## 1. 固定信息 + INFO信息 + FORMAT信息

### 1.1. INFO信息
|name|    解释|
|----|----|
|AC（Allele Count）|    表示该Allele数目； allele count in genotypes, for each ALT allele, in the same order as listed|
|AF（Allele Frequence）|    表示Allele的频率 ；allele frequency for each ALT allele in the same order as listed: use this when estimated from primarydata, not called genotypes|
|AN（Allele Number）|   表示Allele的总数目； : total number of alleles in called genotypes|
|DP|  reads覆盖度，其中有些reads被过滤掉。需要注意的是，这里的DP指的是 combined depth across samples|
|FS|  使用Fisher精确检验来检测strand bias而得到的Phred格式的p值，该值越小越好、
|MQ|  RMS mapping quality|

### 1.2. FORMAT信息
|name|    解释|
|----|----|
|GT（Genotype）|    表示样本基因型，数字之间用“/”隔开。|
|AD（Allele Depth）|    样本中每一种allele的reads覆盖度，中间以逗号隔开，第一个数字表示REF基因型覆盖度，第二个数字表示第一个variant覆盖度，第三个数字表示第二个variant覆盖度（如果有的话，以此类推） AD is the unfiltered allele depth, including reads that did not pass the variant caller’s filters|
|DP（Depth）|   表示样本在该位点的覆盖度，reads with MQ=255 or with bad mates are filtered. DP is the filtered depth, at the sample level.|
|GQ（Genotype Quality）|    基因型的质量值。Phred格式(Phred_scaled)的质量值，表示在该位点该基因型存在的可能性；该值越高，则Genotype的可能性越大；计算方法：Phred值 = -10 * log (1-p) ，其中p为基因型存在的概率。|
|PL|  指定的三种基因型的质量值(provieds the likelihoods of the given genotypes)。这三种指定的基因型为(0/0,0/1,1/1)|，这三种基因型的概率总和为1。和之前不一样的是，该值越大，表明为该种基因型的可能性越小。 Phred值 = -10 * log §，其中p为基因型存在的概率。How much less likely that genotype is compared to the best one|

### 1.3. INFO_
#### 1.3.1. **QualByDepth (QD)**
This is the variant confidence (from the QUAL field) divided by the unfiltered depth of non-reference samples.

#### 1.3.2. **FisherStrand (FS)**
Phred-scaled p-value using Fisher’s Exact Test to detect strand bias (the variation being seen on only the forward or only the reverse strand) in the reads. More bias is indicative of false positive calls.

#### 1.3.3. **RMSMappingQuality (MQ)**
This is the Root Mean Square of the mapping quality of the reads across all samples.

#### 1.3.4. **MappingQualityRankSumTest (MQRankSum)**
This is the u-based z-approximation from the Mann-Whitney Rank Sum Test for mapping qualities (reads with ref bases vs. those with the alternate allele). Note that the mapping quality rank sum test can not be calculated for sites without a mixture of reads showing both the reference and alternate alleles, i.e. this will only be applied to heterozygous calls.

#### 1.3.5. **ReadPosRankSumTest (ReadPosRankSum)**
This is the u-based z-approximation from the Mann-Whitney Rank Sum Test for the distance from the end of the read for reads with the alternate allele. If the alternate allele is only seen near the ends of reads, this is indicative of error. Note that the read position rank sum test can not be calculated for sites without a mixture of reads showing both the reference and alternate alleles, i.e. this will only be applied to heterozygous calls.

#### 1.3.6. **StrandOddsRatio (SOR)**
The StrandOddsRatio annotation is one of several methods that aims to evaluate whether there is strand bias in the data. Higher values indicate more strand bias.
+ Biallelic site
二等位基因位点
+ Multiallelic sites
多等位基因位点

### 1.4. FORMAT_

#### 1.4.1. 
The sample-level DP is in some sense reflective of the power I have to determine the genotype of the sample at this site, while the AD tells me how many times I saw each of the REF and ALT alleles in the reads, free of any bias potentially introduced by filtering the reads. If, for example, I believe there really is a an A/T polymorphism at a site, then I would like to know the counts of A and T bases in this sample, even for reads with poor mapping quality that would normally be excluded from the statistical calculations going into GQ and QUAL.

Note that because the AD includes reads and bases that were filtered by the caller (and in case of indels, is based on a statistical computation), it should not be used to make assumptions about the genotype that it is associated with. Ultimately, the phred-scaled genotype likelihoods (PLs) are what determines the genotype calls.


https://software.broadinstitute.org/gatk/documentation/article.php?id=4721

**AlleleBalance**

##### 1.4.1.1. DepthPerAlleleBySample (AD)
Depth of coverage of each allele per sample
https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_annotator_DepthPerAlleleBySample.php

##### 1.4.1.2. Coverage (DP)
Total depth of coverage per sample and over all samples.

These two values will usually, but not always sum to the DP value. Reads that are not used for calling are not counted in the DP measure, but are included in AD.
+  AD is the unfiltered allele depth, i.e. the number of reads that support each of the reported alleles. 
+ DP is the filtered depth, at the sample level. 

https://software.broadinstitute.org/gatk/documentation/article.php?id=1268
Total depth of coverage per sample and over all samples.

https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_annotator_Coverage.php

##### 1.4.1.3. AlleleBalance
##### 1.4.1.4. AlleleBallanceBySample
##### 1.4.1.5. DepthPerSampleHC
Depth of informative coverage for each sample.
##### 1.4.1.6. StrandAlleleCountsBySample
Number of forward and reverse reads that support each allele
https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_annotator_StrandAlleleCountsBySample.php



**关于两个不同的DP，**
The INFO DP is the combined depth over all samples. The FORMAT DP is for the sample. DP at sample level is for individual samples. At site level in INFO, it should be approximately sum of all sample level DPs

### 1.4.2. GQ vs PL
GQ is the difference between the PL of the second most likely genotype, and the PL of the most likely genotype. the values of the PLs are normalized so that the most likely PL is always 0, so the GQ ends up being equal to the second smallest PL, unless that PL is greater than 99. Basically the GQ gives you the difference between the likelihoods of the two most likely genotypes. If it is low, you can tell there is not much confidence in the genotype.
**QUAL vs GQ**:
The basic difference is that QUAL refers to the variant site whereas GQ refers to a specific sample's GT.

+ QUAL tells you how confident we are that there is some kind of variation at a given site. The variation may be present in one or more samples.

+ GQ tells you how confident we are that the genotype we assigned to a particular sample is correct. It is simply the second lowest PL, because it is the difference between the second lowest PL and the lowest PL (always 0).

#### Ref_Info 
https://software.broadinstitute.org/gatk/documentation/article.php?id=4860

**MQ** is the mapping quality, which is the fifth column in SAM record. QUAL, meanwhile, is the base quality score, which is derived from the 11th column in SAM record.

**MQ** is typically an indication of how unique the region’s sequence is, the higher the MQ, the more unique the sequence. QUAL, is the sequencing quality, which can be platform biased, e.g. Ion seemed to have lower QUAL compared to Illumina.

#### Ref_Info
http://gatkforums.broadinstitute.org/gatk/discussion/6455/biallelic-vs-multiallelic-sites
https://gatkforums.broadinstitute.org/gatk/discussion/1268/what-is-a-vcf-and-how-should-i-interpret-it

## 2. Variants filter
**For SNPs:**
QD < 2.0
MQ < 40.0
FS > 60.0
SOR > 3.0
MQRankSum < -12.5
ReadPosRankSum < -8.0
QUAL < 30.0

**For indels:**
QD < 2.0
ReadPosRankSum < -20.0
InbreedingCoeff < -0.8
FS > 200.0
SOR > 10.0
QUAL < 30.0

### 2.1. bcftools filter & bcftools view
```
# 首先对结果进行index
tabix -p vcf raw_samtools.vcf.gz
# 注意tabix 使用时输入文件一定是使用bgzip压缩的vcf文件

# 分别提取snp和indel结果
bcftools view -i "type='snps'" raw_samtools.vcf.gz -O z\
 -o raw_samtools_snps.vcf.gz && tabix \
 -p vcf raw_samtools_snps.vcf.gz
bcftools view -i "type='indels'" raw_samtools.vcf.gz -O z \
-o raw_samtools_indels.vcf.gz && tabix \
-p vcf raw_samtools_indels.vcf.gz

#
bcftools filter -e "MQ < 40 || QUAL < 30" -s LOWQUAL \
raw_samtools_snps.vcf.gz |bcftools view \
-f PASS - > filtered_samtools_snps.vcf.gz
```

```
#
bcftools norm [OPTIONS] file.vcf.gz

bcftools view -m2 -M2 -v snps input.vcf.gz

perl -lane 'if(/^#/ or length("$F[3]$F[4]")==2){print}' file.vcf

awk '/#/{print;next}{if($5 !~ /,/ && length($5)==1 && length($4)==1){print}}' file.vcf

```

### 2.2. GATK filter
```bash
java -Xmx8g -jar /home/zf/software/GenomeAnalysisTK.jar \
-T VariantFiltration \
-R /mnt/zf/rice_data/all.con.fa \
-V raw_snps_gatk.vcf \
--filterExpression "( vc.hasAttribute('QD') && QD<2.0 ) || FS > 60.0 \
|| MQ < 40.0 || ( vc.hasAttribute('ReadPosRankSum' ) && ReadPosRankSum < -8.0 ) \
|| ( vc.hasAttribute('MQRankSum') && MQRankSum < -12.5 ) \
|| ( vc.hasAttribute('SOR') && SOR > 10.0 ) || QUAL < 30.0 " \
--filterName "my_snp_filter" \
-o filter_snps_gatk.vcf

java -Xmx8g -jar /home/zf/software/GenomeAnalysisTK.jar \
-T SelectVariants \
-R /mnt/zf/rice_data/all.con.fa \
-V filter_snps_gatk.vcf \
-ef -o filtered_snps_gatk.vcf


java -Xmx8g -jar /home/zf/software/GenomeAnalysisTK.jar \
-T VariantFiltration \
-R /mnt/zf/rice_data/all.con.fa \
-V raw_indels_gatk.vcf \
--filterExpression " ( vc.hasAttribute('QD') && QD<2.0 ) \|| FS > 200.0 \
|| ( vc.hasAttribute ('ReadPosRankSum') && ReadPosRankSum < -20.0 ) \
|| ( vc.hasAttribute('SOR') && SOR > 10.0 )|| QUAL < 30.0" \
--filterName "my_indel_filter" \
-o filter_indels_gatk.vcf

java -Xmx8g -jar /home/zf/software/GenomeAnalysisTK.jar \
-T SelectVariants \
-R /mnt/zf/rice_data/all.con.fa \
-V filter_indels_gatk.vcf \
-ef -o filtered_indels_gatk.vcf
```
关于SNP位点的过滤，找了一篇文章，该文章去掉了大于4小于1000的位点，当然可以根据你的需求来过滤。
参考文献：
Li M, Tian S, Yeung C K L, et al. Whole-genome sequencing of Berkshire (European native pig) provides insights into its origin and domestication[J]. Scientific reports, 2014, 4.

```
Filter SNPs with quality minor than 10:
vcftools --vcf var.raw.vcf --recode --recode-INFO-all --minQ 10 --stdout | gzip -c > filter.var.raw.vcf.gz

Filter SNPs with more than 20% of missing data:
vcftools --vcf var.raw.vcf --recode --recode-INFO-all --max-missing 0.8 --stdout 

Select all SNPs in a determiante region:
vcftools --vcf var.raw.vcf --recode --recode-INFO-all --chr CP4_pseudomolecule00 --from-bp 161000 --to-bp 245000 --stdout 

Calculate the observed heterozygosity of all SNPs:
vcftools --vcf var.raw.vcf --het --out ril

Calculate the SNP density:
vcftools --vcf var.raw.vcf --SNPdensity 10000 --out ril
```

### 2.3. freeabyes filter

#### Ref_Info
去除位点的方法参考：https://www.biostars.org/p/141156/
http://blog.sina.com.cn/s/blog_83f77c940102wdl1.html
https://bioinf.comav.upv.es/courses/sequence_analysis/snp_calling.html
https://github.com/vcflib/vcflib#vcffilter
http://gatkforums.broadinstitute.org/gatk/discussion/7376/about-filtering-the-somatic-variants
http://gatkforums.broadinstitute.org/gatk/discussion/2806/howto-apply-hard-filters-to-a-call-set


```r
#install package
source("http://bioconductor.org/biocLite.R")
biocLite("VariantAnnotation")
 
#load library
library(VariantAnnotation)
```

```
eg <- system.file("extdata", "chr22.vcf.gz", package="VariantAnnotation")
vcf <- readVcf(eg, "hg19")
AF <- info(vcf)$AF
summary(AF)
hist(AF, breaks=50, xlab='Alternate Allele Frequency')
```

```
source("http://bioconductor.org/biocLite.R")
biocLite("SNPRelate")
```



## Reference_Info
https://davetang.org/muse/2015/07/24/dna-sequencing-data/https://davetang.org/muse/2015/07/24/dna-sequencing-data/  
http://www.bioconductor.org/packages/release/bioc/vignettes/SNPRelate/inst/doc/SNPRelateTutorial.html
http://kaopubear.top/2017-10-13-variantbasic1.html
https://gatkforums.broadinstitute.org/gatk/discussion/1268/what-is-a-vcf-and-how-should-i-interpret-it